lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -336,11 +342,19 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
     Livebook.Hubs.create_secret(secret)
   end
 
+  defp grant_access(secrets, secret_name, :hub, socket) do
+    secret = Enum.find(secrets, &(&1.name == secret_name and match?({:hub, _}, &1.origin)))
```

Comment:
So if there are multiple secrets with the same name, grant access always refers to the first one? (in non-deterministic order?)

Reply:
the list the component receives from `SessionLive` is already sorted, so yeah, it would be the first one from this order.